src/cljs/kixi/hecuba/tabs/hierarchy/tapestry.cljs

```diff
@@ -0,0 +1,382 @@
+(ns kixi.hecuba.tabs.hierarchy.tapestry
+  (:require-macros [cljs.core.async.macros :refer [go go-loop]]
+                   [hiccups.core :as hiccups])
+  (:require [clojure.string :as str]
+            [cljs.core.async :refer [<! >! chan put!]]
+            [cljs.reader :as reader]
+            [om.core :as om :include-macros true]
+            [ajax.core :refer [GET]]
+            [sablono.core :as html :refer-macros [html]]
+            [cljs-time.core   :as t]
+            [cljs-time.format :as tf]
+            [cljs-time.coerce :as tc]
+            [kixi.hecuba.bootstrap :as bs]
+            [kixi.hecuba.common :refer (log) :as common]
+            [kixi.hecuba.tabs.hierarchy.data :as data]
+            [kixi.hecuba.model :refer (app-model)]
+            [cljs.core.async :refer [<! >! chan close! sliding-buffer put! alts!]]
+            [cljs.reader :refer [read-string]]
+            [om.dom :as dom :include-macros true]
+            [thi.ng.geom.viz.core :as viz]
+            [thi.ng.geom.svg.core :as svg]
+            [thi.ng.geom.core.vector :as v]
+            [thi.ng.geom.core :as g]
+            [thi.ng.geom.core.utils :as gu]
+            [thi.ng.math.simplexnoise :as n]
+            [thi.ng.math.core :as m :refer [PI]]
+            [thi.ng.color.gradients :as grad]
+            [thi.ng.ndarray.core :as nd]
+            [hiccups.runtime :as hiccupsrt]
+            [goog.string :as gstring]
+            [goog.string.format]
+            [bardo.interpolate :refer [interpolate pipeline]]
+            [kixi.hecuba.widgets.datetimepicker-small :as dtp]
+            [kixi.hecuba.tabs.hierarchy.raw-data :as raw]))
+
+
+(def x-readings 14)
+(def y-readings 24)
+(def chart-width (atom 800))
+(def chart-height (atom 600))
+(def chart-data-chan (atom nil))
+(def chart-fill-oor-cells? (atom true))
+(def default-gradations 20)
+(def invalid-cell-colour "#555555")
+
+(def colour-scheme
+  "http://colorbrewer2.org/?type=diverging&scheme=RdYlBu&n=10"
+  [[49 54 149] [69 117 180] [116 173 209] [171 217 233] [224 243 248] [254 224 144] [253 174 97] [244 109 67] [215 48 39] [165 0 38]])
```

Comment:
I think this is available in https://github.com/thi-ng/color but this works for here.